### PR TITLE
Fixed Kusto change connections for previous notebooks

### PR DESCRIPTION
**Pull request by VasuBhog**
This PR fixes #12178 by using currentKernelAlias instead of constants. Should load all connections available when the notebook is not connected to anything. 
image

### CODE REVIEW

File: src/sql/workbench/contrib/notebook/browser/notebookActions.ts [hunk 1]
```diff
@@ -455,10 +455,16 @@ export class AttachToDropdown extends SelectBox {
 	 **/
 	public async openConnectionDialog(useProfile: boolean = false): Promise<boolean> {
 		try {
+			//Get all providers to show all available connections in connection dialog
+			let providers = [];
+			providers = this.model.getApplicableConnectionProviderIds(this.model.clientSession.kernel.name);
```
Comment (chlafreniere) [inline src/sql/workbench/contrib/notebook/browser/notebookActions.ts:460]: if you're setting this here, can just do

`let providers = this.model.`etc

Comment (chlafreniere) [reply src/sql/workbench/contrib/notebook/browser/notebookActions.ts:460]: does `getApplicableConnectionProviderIds` return an empty array, or undefined?

File: src/sql/workbench/services/notebook/browser/models/notebookModel.ts [hunk 2]
```diff
@@ -640,7 +642,7 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			let standardKernels = find(this._standardKernels, kernel => this._defaultKernel && kernel.displayName === this._defaultKernel.display_name);
 			let connectionProviderIds = standardKernels ? standardKernels.connectionProviderIds : undefined;
 			let providerFeatures = this._capabilitiesService.getCapabilities(profile.providerName);
-			if (connectionProviderIds.length > 0) {
+			if (connectionProviderIds.length > 0 && this._currentKernelAlias) {
```
Comment (chlafreniere) [inline src/sql/workbench/services/notebook/browser/models/notebookModel.ts:645]: I don't quite understand this logic. Are we setting `this._currentKernelAlias` in another place as well?

File: src/sql/workbench/services/notebook/browser/models/notebookModel.ts [hunk 3]
```diff
@@ -593,8 +593,10 @@ export class NotebookModel extends Disposable implements INotebookModel {
 			let providers = this._capabilitiesService.providers;
 			for (const server in providers) {
 				let alias = providers[server].connection.notebookKernelAlias;
+				//Add Notebook Kernel Alias to kernelAliases
```
Comment (chlafreniere) [inline src/sql/workbench/services/notebook/browser/models/notebookModel.ts:596]: Nit: Generally we have a space at the start of a comment

File: src/sql/workbench/contrib/notebook/browser/notebookActions.ts (hunk 1 above)
Comment (VasuBhog) [reply src/sql/workbench/contrib/notebook/browser/notebookActions.ts:460]: Yes, `getApplicableConnectionProviderIds` would return undefined if their arent any provider ids.

File: src/sql/workbench/contrib/notebook/browser/notebookActions.ts [hunk 4]
```diff
@@ -455,10 +455,15 @@ export class AttachToDropdown extends SelectBox {
...
+			// Get all providers to show all available connections in connection dialog
+			let providers = this.model.getApplicableConnectionProviderIds(this.model.clientSession.kernel.name);
+			for (let alias of this.model.kernelAliases) {
+				providers = providers.concat(this.model.getApplicableConnectionProviderIds(alias));
+			}
 			let connection = await this._connectionDialogService.openDialogAndWait(this._connectionManagementService,
 				{
 					connectionType: ConnectionType.temporary,
-					providers: this.model.getApplicableConnectionProviderIds(this.model.clientSession.kernel.name)
+					providers: providers,
```
Comment (chlafreniere) [inline src/sql/workbench/contrib/notebook/browser/notebookActions.ts:466]: don't need comma

**Review by chlafreniere — APPROVED** [review]
One tiny thing, but otherwise looks good.

**Review by JustinMDotNet — APPROVED** [review]
Looks great!

**Pull request merged**